src/core.jl

```diff
@@ -148,38 +202,84 @@ function fit{C}(m::KernelNB, X::Matrix, y::Vector{C})
     return m
 end
 
+
 """
     sum_log_x_given_c!(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::KernelNB, X::Matrix, c)
 
 Updates input vector `class_prob` with the sum(log(P(x|C)))
 """
-function sum_log_x_given_c!(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::KernelNB, X::Matrix, c)
+function sum_log_x_given_c!(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::KernelNB, X::Matrix, c)# TODO: depricate
     for i in 1:size(X, 2)  # for each sample
-        for j in 1:size(X, 1)  # for each class
+        for j in 1:size(X, 1)  # for each feature
             feature_prob[j] = pdf(m.c_kdes[c][j], X[j, i])
         end
         class_prob[i] = sum(log(feature_prob))
     end
 end
 
+
+"""computes log[P(x⃗ⁿ|c)] ≈ ∑ᵢ log[p(xⁿᵢ|c)] """
+function sum_log_x_given_c!{T <: AbstractFloat, U <: Int}(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::HybridNB, continuous_features::Vector{Vector{T}}, discrete_features::Vector{Vector{U}}, c)
```

Comment:
Maybe. Initial design was to split types and functions into different files for easier work in REPL (parametric types still result in an error when reevaluated). Now we have more NB types, so splitting stuff into something like `common.jl`, `multinomial.jl`, `gaussian.jl` and `hybrid.jl` sounds reasonable.